src/App.jsx

```diff
@@ -14,33 +17,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_TYPE_ALPHABETICALLY = 'alphabetically';
+const SORT_TYPE_BY_LENGTH = 'by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const goods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortType) {
+    goods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_TYPE_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SORT_TYPE_BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortType(SORT_TYPE_ALPHABETICALLY)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SORT_TYPE_ALPHABETICALLY,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortType(SORT_TYPE_BY_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SORT_TYPE_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setReverse(!reverse)}
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverse,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(sortType || reverse) && (
+          <button
+            onClick={() => {
+              setSortType('');
+              setReverse(false);
+            }}
```

Comment:
Move this logic from jsx to the helper variable and use it here
```suggestion
            onClick={reset}
```